app/code/Magento/InventoryInStorePickup/Plugin/InventoryApi/SourceRepository/LoadInStorePickupOnGetListPlugin.php

```diff
@@ -20,24 +20,18 @@ class LoadInStorePickupOnGetListPlugin
      * @param SourceSearchResultsInterface $sourceSearchResults
      *
      * @return SourceSearchResultsInterface
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function afterGetList(
         SourceRepositoryInterface $subject,
         SourceSearchResultsInterface $sourceSearchResults
     ):SourceSearchResultsInterface {
-        $sources = [];
-
         foreach ($sourceSearchResults->getItems() as $source) {
-            $pickupAvailable = $source->getData(InStorePickupInterface::IN_STORE_PICKUP_CODE);
-
             $extensionAttributes = $source->getExtensionAttributes();
-            $extensionAttributes->setInStorePickup($pickupAvailable);
-
-            $source->setExtensionAttributes($extensionAttributes);
 
-            $sources[] = $source;
+            $pickupAvailable = $source->getData(PickupLocationInterface::IS_PICKUP_LOCATION_ACTIVE);
+            $extensionAttributes->setIsPickupLocationActive((bool)$pickupAvailable);
```

Comment:
According to the `SourceInterface` contract `getExtensionAttributes` may also return null.

Reply:
Added check for `null`.